### DISCUSSION TITLE
Question: Optimizing C#

### DISCUSSION THREAD

**Discussion by andrewbabbittdev** [PlummersSoftwareLLC/Primes#519] [Q&A]
Looking to do some testing with some upcoming experimental AOT work in .NET. These would be optimizations that could be used on all the existing solutions that are in play. Should I update all of the existing solutions to use the experimental AOT compiler or should I make new modified solutions instead?

**Comment by rbergen** [comment]
Converting this to a Q&A discussion topic.

**Comment by rbergen** [comment]
In principle, performance improvements to existing solutions need to be submitted as PRs that modify the solutions in question. If I'm honest, I would prefer that to be the approach used here as well. That does mean that you would need to try and get approval from the various solution authors to apply this change. That's the approach we use in general anyway, but I think it's even more important now, considering the fact that it concerns an experimental compiler, I do think it is primarily up to the original authors to decide how comfortable they are with applying this to their solutions.

If none of the original authors agree with adopting the AOT compiler, then I will consider you applying it to one existing solution in a separate/new solution, provided there clearly is an interesting story to show concerning its performance. I'd like to add to that that we've never yet removed previously merged solutions that have become moot due to subsequent developments, but it crossed my mind that this could become a consideration for such a separate solution. Specifically, this could happen if and when the currently existing .NET solutions adopt the AOT compiler, if and when it becomes mainstream and generally adopted.
